protocol/avdl/stellar1/bundle.avdl

```diff
@@ -15,48 +21,155 @@ protocol bundle {
+    array<BundleVisibleEntryV1> accounts;
   }
 
   record BundleSecretV1 {
     Hash visibleHash; // SHA256 of msgpack(BundleVisibleV1)
-    array<BundleSecretEntry> accounts;
+    array<BundleSecretEntryV1> accounts;
   }
 
-  enum AccountMode {
-    NONE_0,
-    USER_1 // Each of the user's devices has access to the keys
+  // V1 Server-visible attributes of an account.
+  record BundleVisibleEntryV1 {
+    AccountID accountID;
+    AccountMode mode;
+    boolean isPrimary;        // whether this is the primary account (public)
+  }
+
+  // V1 Secret attributes of an account.
+  record BundleSecretEntryV1 {
+    AccountID accountID;
+    array<SecretKey> signers;
+    string name;
   }
 
-  // Server-visible attributes of an account.
-  record BundleVisibleEntry {
+  // --------------------------------------------------------------------------
+  // V2
+  // --------------------------------------------------------------------------
+
+  record BundleVisibleV2 {
+    BundleRevision revision;
+    Hash prev; // SHA256 of previous msgpack(EncryptedBundle)
+    array<BundleVisibleEntryV2> accounts;
+  }
+  
+  record BundleSecretV2 {
+    Hash visibleHash; // SHA256 of msgpack(BundleVisibleV2)
+    array<BundleSecretEntryV2> accounts;
+  }
+
+  // V2 Server-visible attributes of an account.
+  record BundleVisibleEntryV2 {
     AccountID accountID;
     AccountMode mode;
-    boolean isPrimary; // whether this is the primary account (public)
+    boolean isPrimary;        // whether this is the primary account (public)
+    BundleRevision acctBundleRevision; // revision of AccountBundleSecret
+    Hash encAcctBundleHash;   // SHA256 of msgpack(EncryptedAccountBundle) for this account
+  }
+  
+  // V2 Secret attributes of an account.
+  record BundleSecretEntryV2 {
+    AccountID accountID; // this is here to check that the bundle isn't corrupt
+    string name;
   }
 
-  // Secret attributes of an account.
-  record BundleSecretEntry {
+  // --------------------------------------------------------------------------
+  // Unsupported 
+  // --------------------------------------------------------------------------
+
+  record BundleSecretUnsupported {}
+
+  // ==========================================================================
+  // AccountBundles
+  // ==========================================================================
+
+  // The same format as in chat1.EncryptedData (and KBFS, Git)
+  record EncryptedAccountBundle {
+    int   v;                           // Version of the CRYPT.
+    bytes e;                           // encrypted msgpacked AccountBundleSecretVersioned (output of secretbox)
+    keybase1.BoxNonce n;               // nonce
+    keybase1.PerUserKeyGeneration gen; // PUK generation that was used
+  }
+
+  // Version of the AccountBundleSecret data.
+  enum AccountBundleVersion {
+    V1_1,
+    V2_2,
+    V3_3,
+    V4_4,
+    V5_5,
+    V6_6,
+    V7_7,
+    V8_8,
+    V9_9,
+    V10_10
+  }
```

Comment:
I bet the account bundle versioning will be pretty intimately tied to BundleVersion so it might make sense to use `BundleVersion`. But this is fine too